### PR TITLE
Fixed issue in config messing ConfigResourceFactory with ResourceFactory

### PR DESCRIPTION
**Pull request by georgique**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

There is a bit of a mess in the module config which points to some non-existing classes. Causes issues in some other modules.

### CODE REVIEW

**Comment by georgique** [comment]
I thinks we should also fix #5 here, but I need a hint on how it's supposed to work

File: config/module.config.php [hunk 1]
```diff
@@ -6,6 +6,8 @@
 
 namespace Laminas\ApiTools\Configuration;
 
+use LaminasTest\ApiTools\Configuration\TestAsset\ConfigWriter;
```
Comment (froschdesign) [inline config/module.config.php]: The usage of a test class is wrong here.

**Comment by froschdesign** [comment]
Please check also the [contributing guidelines](https://github.com/laminas-api-tools/.github/blob/5b3e0cb814531070652af6715feb817f266f5b20/CONTRIBUTING.md) (the link can be found on different places here on GitHub, e.g. under each text field) for the correct target branch.
Thanks in advance! 👍🏻

**Comment by georgique** [comment]
> Please check also the [contributing guidelines](https://github.com/laminas-api-tools/.github/blob/5b3e0cb814531070652af6715feb817f266f5b20/CONTRIBUTING.md) (the link can be found on different places here on GitHub, e.g. under each text field) for the correct target branch. Thanks in advance! 👍🏻

It says to target `master` branch, but there is none, so I just targeted the default one. If that's not correct, please let me know which one exactly I need to target.

**Comment by froschdesign** [comment]
@georgique 
> It says to target master branch…

Sorry, the anchor is missing in the link:

> #### What branch to issue the pull request against?
> 
> Which branch should you issue a pull request against?
> 
> - For fixes against the stable release, issue the pull request against the release branch matching the minor release you want to fix.
> - For new features, or fixes that introduce new elements to the public API (such as new public methods or properties), issue the pull request against the default branch.

https://github.com/laminas-api-tools/.github/blob/5b3e0cb814531070652af6715feb817f266f5b20/CONTRIBUTING.md#what-branch-to-issue-the-pull-request-against

This means for this repository:

- the current stable release is [1.6.0](https://github.com/laminas-api-tools/api-tools-configuration/releases)
- and the release / target branch is [1.6.x](https://github.com/laminas-api-tools/api-tools-configuration/tree/1.6.x)

File: config/module.config.php [hunk 2]
```diff
@@ -20,16 +23,16 @@
     'service_manager'         => [
         // Legacy Zend Framework aliases
         'aliases'   => [
-            \ZF\Configuration\ConfigResource::class        => ConfigResource::class,
-            \ZF\Configuration\ConfigResourceFactory::class => ConfigResourceFactory::class,
-            \ZF\Configuration\ConfigWriter::class          => ConfigWriter::class,
-            \ZF\Configuration\ModuleUtils::class           => ModuleUtils::class,
+            \ZF\Configuration\ConfigResource::class  => ConfigResource::class,
+            \ZF\Configuration\ResourceFactory::class => ResourceFactory::class,
```
Comment (georgique) [inline config/module.config.php:27]: Naming is a bit messy there, and that is the reason for this request. Your code snippet above shows aliases, not factories section, so aliases should point at the services, not factories. `ConfigResource` and `ResourceFactory` are services, `ConfigResourceFactory` and `ResourceFactoryFactory` are factories.

image

**Comment by froschdesign** [comment]
@georgique 
> Causes issues in some other modules.

Please let us clarify which modules are affected before anything is changed or a BC break is introduced. _All the old stuff is quite confusing._

**Comment by georgique** [comment]
> This means for this repository:
> 
> the current stable release is [1.6.0](https://github.com/laminas-api-tools/api-tools-configuration/releases)
> and the release / target branch is [1.6.x](https://github.com/laminas-api-tools/api-tools-configuration/tree/1.6.x)

Got it, my bad.
Should we close this PR and I start a new one? I'm not sure that simply changing target will be correct now.

As for the `ConfigWriter` part - I'm switching to `WriterInterface`, which I think makes the most sense now. Some fixes will be required in the other libraries, but I'm not sure it this should be considered a breaking change because those other libraries already have non-working code.

**Comment by georgique** [comment]
@froschdesign I've created another PR #21, which is proper branch + fixed ConfigWriter issue differently so it doesn't break anything. Let's close this one.

**Comment by froschdesign** [comment]
@georgique 
> Let's close this one.

You can do that yourself, because after all, you opened the pull request.

**Comment by georgique** [comment]
> You can do that yourself, because after all, you opened the pull request.

Yes, just was thinking to get your confirmation as a reviewer.

**Comment by froschdesign** [comment]
@georgique 
It is _your_ pull request, so you can close it at any time. 😃

**Pull request closed without merging**